### PR TITLE
Vault Timeout Policy

### PR DESCRIPTION
**Pull request by Hinton**
## Objective
Implements a *Vault Timeout Policy* which lets organizations restrict the maximum vault timeout. It also adds the new `VaultTimeoutInputComponent` which supports `Custom` timeouts.

The `VaultTimeoutInputComponent` is implemented using the `ControlValueAccessor` and `Validator` which provides a nice abstraction layer to encapsulate the complex logic of *Custom* timeouts. The bound model returns a nullable numeric value equal to the timeout in minutes. If the timeout exceeds the allowed timeout the `Validator` will trigger which sets the status to `valid = false`.

I also removed the deprecated `lockTimeout` from `PlatformUtilsService`.

### CODE REVIEW

File: common/src/services/vaultTimeout.service.ts [hunk 1]
```diff
@@ -141,6 +138,25 @@ export class VaultTimeoutService implements VaultTimeoutServiceAbstraction {
...
+
+        if (await this.policyService.policyAppliesToUser(PolicyType.MaximumVaultTimeout)) {
+            const policy = await this.policyService.getAll(PolicyType.MaximumVaultTimeout);
+            // Remove negative values, and ensure it's smaller than maximum allowed value according to policy
+            const timeout = Math.max(0, Math.min(vaultTimeout, policy[0].data.minutes));
+
+            // We really shouldn't need to set the value here, but multiple services relies on this value being correct.
+            if (vaultTimeout != timeout) {
+                await this.storageService.save(ConstantsService.vaultTimeoutKey, timeout);
+            }
```
Comment (Hinton) [inline common/src/services/vaultTimeout.service.ts:152]: I really don't like this, but the dependency chain for `VaultTimeoutService` would need to be reversed to clean this up. Which feels out of scope for this issue.

**Review by MGibson1 — APPROVED** [review]
This looks good to me
